core/src/preferences/PreferencesBackend.cpp

```diff
@@ -194,17 +194,47 @@ namespace ledger {
             unsetEncryption();
 
             auto emptySalt = std::string("");
-            auto pref = getPreferences("__core");
-            auto salt = pref->getString(ENCRYPTION_SALT_KEY, emptySalt);
+            auto saltKey = std::vector<uint8_t>(ENCRYPTION_SALT_KEY.cbegin(), ENCRYPTION_SALT_KEY.cend());
+            auto salt = get(saltKey).value_or(emptySalt);
 
             if (salt == emptySalt) {
-                // we don’t have a proper salt; create one and persist it for future use
+                leveldb::WriteBatch batch;
+
+                // we don’t have a proper salt; create one
                 salt = createNewSalt(rng);
-                pref->editor()->putString(ENCRYPTION_SALT_KEY, salt)->commit();
-            }
 
-            // create the AES cipher
-            _cipher = AESCipher(rng, password, salt, PBKDF2_ITERS);
+                // create the AES cipher
+                _cipher = AESCipher(rng, password, salt, PBKDF2_ITERS);
+
+                // add the salt to the batch to be written
+                auto saltData = std::vector<uint8_t>(salt.cbegin(), salt.cend());
+                auto noCipher = Option<AESCipher>::NONE;
```

Comment:
Make noCipher static

Reply:
Same remark as above. That function is likely to be called more than twice and will be called at initialization of a wallet / application / whatever. I might be wrong, but keeping around in memory an `Option<AESCipher>::NONE` seems overkill to me.

Reply:
Same argument as I put bellow, this is like a constant. We create enums, for a good reason, here it is the same, how many bytes this is? + the function will work faster :) ( this arguments is the same powerful as the one about memory)